src/Lookup/MultiplexingEntityLookup.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\EntityDocument;
+use Wikibase\DataModel\Entity\EntityId;
+use Wikimedia\Assert\Assert;
+use Wikimedia\Assert\ParameterAssertionException;
+use Wikimedia\Assert\ParameterElementTypeException;
+
+/**
+ * Wrapper around repository-specific EntityLookups
+ * picking up the right lookup object for the particular input.
+ *
+ * @package Wikibase\DataModel\Services\Lookup
+ */
+class MultiplexingEntityLookup implements EntityLookup {
+
+	/**
+	 * @var EntityLookup[]
+	 */
+	private $lookups;
+
+	/**
+	 * @param EntityLookup[] $lookups associative array with repository names (strings) as strings
+	 *                                and EntityLookup objects as values.
+	 *
+	 * @throws ParameterAssertionException
+	 * @throws ParameterElementTypeException
+	 */
+	public function __construct( array $lookups ) {
+		Assert::parameterElementType( EntityLookup::class, $lookups, '$lookups' );
+		Assert::parameterElementType( 'string', array_keys( $lookups ), 'array_keys( $lookups )' );
+		foreach ( array_keys( $lookups) as $repositoryName ) {
+			Assert::parameter( strpos( $repositoryName, ':' ) === false, 'array_keys( $lookups )', 'must not contain colons' );
+		}
+		$this->lookups = $lookups;
+	}
+
+	/**
+	 * Returns the entity with the provided id.
+	 *
+	 * @param EntityId $entityId
+	 *
+	 * @return null|EntityDocument
+	 * @throws EntityLookupException
+	 */
+	public function getEntity( EntityId $entityId ) {
+		$repo = $entityId->getRepositoryName();
+		if ( !isset( $this->lookups[$repo] ) ) {
+			throw new EntityLookupException( $entityId, 'Unknown repository: ' . $repo );
+		}
+		return $this->lookups[$repo]->getEntity( $entityId );
```

Comment:
Might be a matter of personal preference but I think it would read better with a private method that takes care of picking the correct repository which can be used here and in `hasEntity` below.